crates/hyfetch/build.rs

```diff
@@ -0,0 +1,173 @@
+use std::path::Path;
+use std::{env, fs};
+
+use indexmap::IndexMap;
+use regex::Regex;
+use unicode_normalization::UnicodeNormalization;
+
+#[derive(Debug)]
+struct AsciiDistro {
+    pattern: String,
+    art: String,
+}
+
+impl AsciiDistro {
+    fn friendly_name(&self) -> String {
+        self.pattern
+            .split('|')
+            .next()
+            .expect("invalid distro pattern")
+            .trim_matches(|c: char| c.is_ascii_punctuation() || c == ' ')
+            .replace(['"', '*'], "")
+    }
+}
+
+fn main() {
+    let neofetch_path = Path::new(env!("CARGO_WORKSPACE_DIR")).join("neofetch");
+
+    println!("cargo:rerun-if-changed={}", neofetch_path.display());
+
+    let out_dir = env::var("OUT_DIR").unwrap();
+    let out_path = Path::new(&out_dir);
+
+    export_distros(neofetch_path, out_path);
+}
+
+fn export_distros<P>(neofetch_path: P, out_path: &Path)
+where
+    P: AsRef<Path>,
+{
+    let distros = parse_ascii_distros(neofetch_path);
+    let mut variants = IndexMap::with_capacity(distros.len());
+
+    for distro in &distros {
+        let variant = distro
+            .friendly_name()
+            .replace(|c: char| c.is_ascii_punctuation() || c == ' ', "_")
+            .nfc()
+            .collect::<String>();
+        variants.insert(variant, distro);
+    }
+
+    let mut buf = "
+#[derive(Clone, Eq, PartialEq, Hash, Debug)]
+pub enum Distro {
+"
+    .to_owned();
+
+    for (variant, distro) in &variants {
+        buf.push_str(&format!(
+            "
+    // {})
+    {variant},
+",
+            distro.pattern
+        ));
+    }
+
+    buf.push_str(
+        "
+}
+",
+    );
+
+    buf.push_str(
+        "
+impl Distro {
+    pub fn ascii_art(&self) -> &str {
+        (match self {
+",
+    );
+
+    let quotes = "#".repeat(80);
+    for (variant, distro) in &variants {
+        buf.push_str(&format!(
+            "
+            Self::{variant} => r{quotes}\"
+{}
+\"{quotes},
+",
+            distro.art
+        ));
+    }
+
+    buf.push_str(
+        "
+        })
+        .trim()
```

Comment:
This is wrong. It should only trim exactly 1 newline from start and end. Over-trimming end is probably fine, but it could potentially ruin the layout if we overtrim start.